presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionExtractor.java

```diff
@@ -25,13 +27,22 @@
 import com.google.common.collect.ImmutableList;
 
 import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
 
 public class ExpressionExtractor
 {
     public static List<Expression> extractExpressions(PlanNode plan)
     {
+        return extractExpressions(plan, n -> n);
```

Comment:
`extractExpressions` without `Lookup` given should not be used for iterative plans.

I rethought that though and I'm falling back to what I introduced already - relying on `Optional<Lookup>` in `ExpressionExtractor.Visitor`. This does not require any special handling in `Lookup` despite what's there already.